tests/unit/lms/services/oauth1_test.py

```diff
@@ -43,6 +46,132 @@ def test_we_can_be_used_to_sign_a_request(self, service, application_instance):
         # one is present.
         assert "oauth_signature=" in auth_header
 
+    def test_sign(
```

Comment:
I made two style of test for this.

This one, mock based and the following one that is more of a functional test for the function checking the output for some canned responses.

Reply:
What's the benefit of each style of test? My intuition would be to _not_ mock a bunch of standard library and third-party encryption libs, and to test the signing by verifying the signed object.